test/helpers/oc/exec.go

```diff
@@ -0,0 +1,139 @@
+package oc
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Execer is interface for collecting arguments for Exec command
+type Execer interface {
+	Command
+
+	// argument for option --config
+	WithConfig(string) Execer
+	// argument for option -n
+	WithNamespace(string) Execer
+
+	// argument for podname
+	Pod(string) Execer
+	// a command can be composed to get the pod name. e.g. oc.Get or oc.Literal
+	WithPodGetter(Command) Execer
+
+	// argument for option -c
+	Container(string) Execer
+	// a command can be composed to get the container name. e.g. oc.Get or oc.Literal
+	WithContainerGetter(Command) Execer
+
+	// command and argument to be executed in pod
+	// arguments for option --
+	WithCmd(string, ...string) Execer
+}
+
+type exec struct {
```

Comment:
@alanconway I think we should consider refraining from introducing command specific structs so that we can stay away from creating a bunch of these to wrapper oc; I suspect we will have a very limited number of them.  The code in origin has a `cli` struct that keeps all the bits needed:  subcommand, args, global options (e.g. namespace).  It then uses them to build up the final call.  

I want to make sure we are not introducing too much here

Reply:
we could create a common struct and embed it in type specific structs

Reply:
@vimalk78 I spoke with @alanconway and he is good with your approach.  Ignore my feedback